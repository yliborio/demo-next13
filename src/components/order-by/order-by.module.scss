@import "../../styles/variables.scss";

.container {
  display: flex;
  flex-direction: column;
  gap: 1rem;
  align-self: flex-end;

  .orderby {
    display: flex;
    justify-content: center;
    cursor: pointer;
    margin-right: 3rem;
    width: max-content;
  }

  .modal {
    position: absolute;
    top: 8rem;
    right: 5rem;
    z-index: 1;
    border: solid 1px $bg-primary;
    background-color: $bg-secondary;
    width: fit-content;
    border-radius: 8px;
    padding: 1rem 2rem;
    font-size: $font-size-3;
    > ul {
      list-style: none;
      > li {
        > a {
          cursor: pointer;
        }

        padding: 4px 0;
      }
    }
  }
}
