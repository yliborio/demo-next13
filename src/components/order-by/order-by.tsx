"use client";
import { useState } from "react";
import styles from "./order-by.module.scss";
import { ArrowDownIcon } from "../icons/arrow-down";
import { useFilterStore } from "core/hooks/useFilter/useFilter";
import { Order } from "core/types/order";

export const OrderBy = () => {
  const [showModal, setShowModal] = useState(false);
  const { setOrder } = useFilterStore();

  const handleChange = (order: Order) => {
    setOrder(order);
    setShowModal(false);
  };

  return (
    <div className={styles["container"]}>
      <div
        className={styles["orderby"]}
        onClick={() => setShowModal(!showModal)}
      >
        <span>Order by</span>
        <ArrowDownIcon />
      </div>
      {showModal && (
        <div className={styles["modal"]}>
          <ul>
            <li>
              <a
                tabIndex={0}
                role="button"
                onClick={() => handleChange(Order.DEFAULT)}
              >
                Featured
              </a>
            </li>
            <li>
              <a
                tabIndex={0}
                role="button"
                onClick={() => handleChange(Order.ASC)}
              >
                Price: Low to High
              </a>
            </li>
            <li>
              <a
                tabIndex={0}
                role="button"
                onClick={() => handleChange(Order.DSC)}
              >
                Price: High to Low
              </a>
            </li>
            <li>
              <a
                tabIndex={0}
                role="button"
                onClick={() => handleChange(Order.RATING)}
              >
                Avg. Customer Review
              </a>
            </li>
          </ul>
        </div>
      )}
    </div>
  );
};
