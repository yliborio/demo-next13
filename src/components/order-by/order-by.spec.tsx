import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { OrderBy } from "./order-by"; // Import your OrderBy component

test("OrderBy component works correctly", () => {
  const { asFragment } = render(<OrderBy />);

  // should match snapshot
  expect(asFragment()).toMatchSnapshot();

  // Check if the initial state hides the modal
  expect(screen.queryByText("Featured")).toBeNull();

  // Click on the "Order by" button to open the modal
  const orderByButton = screen.getByText("Order by");
  fireEvent.click(orderByButton);

  // Check if the modal is displayed after clicking the button
  const featuredOption = screen.getByText("Featured");
  expect(featuredOption).toBeInTheDocument();

  // Click on the "Featured" option
  fireEvent.click(featuredOption);

  // Check if the modal is closed
  expect(screen.queryByText("Featured")).toBeNull();

  // Test keydown events
  fireEvent.keyDown(orderByButton, { code: "Enter" });
  expect(screen.queryByText("Featured")).toBeInTheDocument();

  fireEvent.keyDown(screen.getByText("Featured"), { code: "Enter" });
  expect(screen.queryByText("Featured")).toBeNull();
});
