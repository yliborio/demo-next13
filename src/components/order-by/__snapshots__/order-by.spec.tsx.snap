// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderBy component works correctly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="orderby"
      role="button"
      tabindex="0"
    >
      <span>
        Order by
      </span>
      <svg
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="12"
          x2="12"
          y1="5"
          y2="19"
        />
        <polyline
          points="19 12 12 19 5 12"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
